### ISSUE TITLE
Filebot not found

### ISSUE THREAD

**Issue by fexigo** [Laharah/deluge-FileBotTool#32]
Hi

I got an windows 7 computer with the latest version of Deluge, Java, Filebot and FilebotTool plugin for Deluge. But I still get the message "Filebot not found". Filebot -version is working, and I can start Filebot. 

What am I doing wrong??

**Comment by Laharah** [comment]
No problem, this usually boils down to one of a few things on windows:

- Have you tried restarting both deluge and your PC since installing filebot?
- could you tell me where filebot is installed to? It should be something like C:\Program Files\FileBot\FileBot.exe.
- what comes up when you type filebot -version in the command line?
- are the permissions on filebot.exe and deluge.exe/deluged.exe the same?

**Comment by rednoah** [comment]
It's best to just assume that the `filebot` command is in the `%PATH%` because the installation folder may not always be the same for everyone, especially not if deployed via the Windows Store.

**Comment by Laharah** [comment]
@rednoah, Yeah my module starts with checking the path, but it dosen't always work, particularly when permissions and strange path modifications for deluge users come in. I'm on the lookout for more places to check.

**Comment by fexigo** [comment]
Hi.
I have tried to restart and reinstall both Deluge and Filebot.

Filebot is installed at C:\Program Files\FileBot\Filebot.exe

C:\Users\Fexigo>filebot -version
FileBot 4.7.9 (r4984) / Java(TM) SE Runtime Environment 1.8.0_131 / Windows 7 6.
1 (amd64)

Same permission for both programs.

**Comment by Laharah** [comment]
okay, just to make sure it's not the permissions, when you start deluge, can you right click it and hit run as administrator? If filebot still doesn't show up it'll have to be something else.

**Comment by rednoah** [comment]
@Laharah It's kinda tricky on Windows. I'd call commands with `cmd.exe /c command ...` to make sure it works as expected.

Actually, many users somehow end up with fucked up `%PATH%` variables (some installers do that) so it does make sense to check known locations as well.

If FileBot is managed by the Windows Store, then the executable will be here:
```
%USERPROFILE%\AppData\Local\Microsoft\WindowsApps\filebot.exe
```

**Comment by Inglend** [comment]
I'm having the same issue but on Mac OS. "Filebot version not found". I have Filebot installed and Deluge, and Filebot Tool works too. Any ideas?

**Comment by Laharah** [comment]
Sure, can you open up Terminal and tell me the output of these commands?

- `ls -alL $(which filebot)`
- `ls -alL /Applications/Deluge.app/Contents/MacOS`

And can you also tell me a little about how your deluge is setup (eg. is it running on a server or locally)?

**Comment by rednoah** [comment]
On `Mac OS X`, the `filebot` tool should be linked at `/usr/local/bin/filebot`.

**Comment by fexigo** [comment]
Hi.
I tried to start deluge with admin. Still having the same problem.

Also tried to reinstall deluge, filebot multiple times.

**Comment by Laharah** [comment]
@fexigo can you tell me a bit more about your setup? Is deluge in client-server mode or in "classic" mode (you can check by going to preferences -> interface). What version of windows are you on?

**Comment by fexigo** [comment]
Hi. Im using classic mode and its a windows 7 pro computer. 

The strange thing about this is when i right click the torrent and choose Filebot Tool i can see in task manager that two processes of Filebot *32 is started. And if i try to reboot my computer i got a system error that telling me that the application was unable to start correctly (0xc0000000XX).

**Comment by Laharah** [comment]
Hmm, okay, it seems like something else is going on. I've posted a release with some extra debug logging here:

https://github.com/Laharah/deluge-FileBotTool/releases/tag/1.1.9

Can you run it with logging on ([see instructions here](http://dev.deluge-torrent.org/wiki/Troubleshooting#EnableDelugeLogging)) and send me the logs?

**Comment by fexigo** [comment]
Wow! That is strange. Now its working. hmmmmmmm

**Comment by Laharah** [comment]
That is really weird, I only added some logging. Well, if you run into problems, feel free to post back here! I'm going to keep this open incase @Inglend needs more help.

**Comment by fexigo** [comment]
Thank you.

**Comment by fexigo** [comment]
I was a little bit fast. When i launch deluge-debug.exe everythings works. But if i lanuch deluge.exe i still got Filebot not found.

**Comment by Laharah** [comment]
are deluge-debug.exe and deluge.exe both in the same folder? Are you launching them both the same way from the command line?

**Comment by Inglend** [comment]
@Laharah It's a Mac Mini running locally I think. 
![screenshot filebot  @tool](https://cloud.githubusercontent.com/assets/28608047/26153243/ce9f086a-3b0a-11e7-8e15-214fb8f540a0.png)

@rednoah  ok I found the folder up until bin but no filebot in there. How do I link it?

**Comment by rednoah** [comment]
You can install `filebot` correctly by running this command:

```
sh -xu <<< "$(curl -fsSL https://raw.githubusercontent.com/filebot/plugins/master/installer/brew.sh)"
```

**Comment by Inglend** [comment]
Yes now it works! thank you!

**Comment by fexigo** [comment]
Sorry, i have been away for some time. If i start deluge with just using the icon its not working. If i start deluge debug from cmd line it working. I have also tried to start deluge from cmd line without any success

**Comment by Laharah** [comment]
okay this a weird one. Since you can't get debug logs, you'll have to download and install this special debug version I made: 

https://github.com/Laharah/deluge-FileBotTool/releases/tag/1.1.9-debug

It will log the filebot error to the filebot version string (where it normally says "Filebot not found" in the rename dialog). Could you copy what it says to me?

**Comment by thebrianlenz** [comment]
Following all the suggestions posted in this issue, I still run into the "File not Found" problem.

Using your debug release, it shows "[Error 5] Access is denied." What I've found is that upon restarting my computer, there are no issues. Everything launches and doesn't indicate any problems. However, if Deluge is closed and reopened, the problem returns.

Additionally, when Deluge is first opened, a Filebot.exe process is started (which I assume is normal). However, when Deluge is closed, the Filebot.exe process is not ended. I anticipate this could be part of the issue, though ending the process doesn't seem to help. If Deluge is reopened multiple times, multiple Filebot.exe processes stack up.

**Comment by Laharah** [comment]
The filebot launch when starting deluge is just a version check and making sure that filebot.exe can be accessed, it should close almost immediately. With the `Access is denied` error it seems like there is some kind of permissions problem somewhere. Also the fact that the filebot instances aren't closing is troubling.

Frankly, I don't really know what could be causing these problems. The fact that the debug version of deluge works and the normal one doesn't is mystifying enough. The fact that it only runs once after a reboot is also really weird. What happens if you reboot, run deluge, quit, kill the filebot task that's still running, and then relaunch deluge? Does it still give you access denied?

Can you double check the permissions? Check and reset the permissions for the entire `Program Files\Deluge` folder, the `Program Files\FileBot` folder and also do the same for the appdata folders for deluge and filebot (hit `windows key + r` and type in `%Appdata%` to get there). Make sure that the user you run deluge with has read/write/execute everywhere in the folders. The user permissions being messed up and a lock somewhere not being released is all I can really think of.
